### PR TITLE
feat: Connection datasource

### PR DESCRIPTION
**Pull request by sfc-gh-fbudzynski**
## Changes
* Connections Datasource
* Generated Documentation for datasource

## Test Plan

* [x] acceptance tests

## References

https://docs.snowflake.com/en/sql-reference/sql/create-connection

## TODO
* SNOW-1788041

### CODE REVIEW

File: \ [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package datasources
```
Comment (sfc-gh-jcieslak) [inline \:1]: remove this file

File: pkg/datasources/connections.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var connectionsSchema = map[string]*schema.Schema{
+	"like": {
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/connections.go:15]: I think there is a common schema somewhere, please use it.

Comment (sfc-gh-fbudzynski) [reply pkg/datasources/connections.go]: yup, it will be used

File: pkg/datasources/connections_acceptance_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+		CheckDestroy: acc.CheckDestroy(t, resources.PrimaryConnection),
+		Steps: []resource.TestStep{
+			{
+				Config: dataConnections,
+				Check: assert.AssertThat(t,
+					assert.Check(resource.TestCheckResourceAttr("data.snowflake_connections.test", "connections.#", "1")),
+					resourceshowoutputassert.ConnectionShowOutput(t, "snowflake_connection.test").
+						HasName(id.Name()).
+						HasSnowflakeRegion(acc.TestClient().Context.CurrentRegion(t)).
+						HasAccountLocator(acc.TestClient().GetAccountLocator()).
```
Comment (sfc-gh-jmichalak) [inline pkg/datasources/connections_acceptance_test.go:102]: Nit: You're missing `created_on`.

**Review by sfc-gh-jmichalak — COMMENTED** [review]
Please add info about this new data source to the migration guide.

**Review by sfc-gh-jmichalak — APPROVED** [review]

File: pkg/datasources/connections_acceptance_test.go [hunk 4]
```diff
@@ -27,7 +27,7 @@ func connectionsData() string {
 
 func TestAcc_Connections_Minimal(t *testing.T) {
 	// TODO: [SNOW-1002023]: Unskip; Business Critical Snowflake Edition needed
-	_ = testenvs.GetOrSkipTest(t, testenvs.TestFailoverGroups)
+	// _ = testenvs.GetOrSkipTest(t, testenvs.TestFailoverGroups)
```
Comment (sfc-gh-jcieslak) [inline pkg/datasources/connections_acceptance_test.go:30]: Uncomment

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**